test/extended/storage/driver_configuration.go

```diff
@@ -0,0 +1,326 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("restoring original driver config: %+v", originalDriverConfigSpec)
+		clusterCSIDriver.Spec.DriverConfig = *originalDriverConfigSpec
+		_, err = oc.AdminOperatorClient().OperatorV1().ClusterCSIDrivers().Update(context.Background(), clusterCSIDriver, metav1.UpdateOptions{})
+		if err != nil {
+			e2e.Failf("failed to update ClusterCSIDriver: %v", err)
+		}
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Context("snapshot options in clusterCSIDriver", func() {
+		var tests = []struct {
+			name                       string
+			clusterCSIDriverOptions    map[string]int
+			cloudConfigOptions         map[string]int
+			successfulSnapshotsCreated int // Number of snapshots that should be created successfully, 0 to skip.
+		}{
+			{
+				name: "global snapshot option",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["global"]["clusterCSIDriver"]: 4,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["global"]["cloudConf"]: 4,
+				},
+				successfulSnapshotsCreated: 4,
+			},
+			{
+				name:                       "default snapshot limit",
+				clusterCSIDriverOptions:    map[string]int{},
+				cloudConfigOptions:         map[string]int{},
+				successfulSnapshotsCreated: 3,
+			},
+			{
+				name: "vsan snapshot option",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["vsan"]["clusterCSIDriver"]: 4,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["vsan"]["cloudConf"]: 4,
+				},
+				successfulSnapshotsCreated: 0,
+			},
+			{
+				name: "vvol snapshot option",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["vvol"]["clusterCSIDriver"]: 4,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["vvol"]["cloudConf"]: 4,
+				},
+				successfulSnapshotsCreated: 0,
+			},
+			{
+				name: "all snapshot options",
+				clusterCSIDriverOptions: map[string]int{
+					snapshotOptions["global"]["clusterCSIDriver"]: 5,
+					snapshotOptions["vsan"]["clusterCSIDriver"]:   10,
+					snapshotOptions["vvol"]["clusterCSIDriver"]:   15,
+				},
+				cloudConfigOptions: map[string]int{
+					snapshotOptions["global"]["cloudConf"]: 5,
+					snapshotOptions["vsan"]["cloudConf"]:   10,
+					snapshotOptions["vvol"]["cloudConf"]:   15,
+				},
+				successfulSnapshotsCreated: 0,
+			},
+		}
+
+		for _, t := range tests {
+			g.It(fmt.Sprintf("%s", t.name), func() {
+				for option, value := range t.clusterCSIDriverOptions {
+					e2e.Logf("updating %s to %d in clustercsidriver", option, value)
+					setClusterCSIDriverSnapshotOptions(oc, option, value)
+				}
+
+				for option, value := range t.cloudConfigOptions {
+					o.Eventually(func() error {
+						return loadAndCheckCloudConf(oc, "Snapshot", option, value)
+					}, time.Minute, time.Second).Should(o.Succeed())
+				}
+
+				if t.successfulSnapshotsCreated > 0 {
+					pvc, err := createTestPVC(oc, oc.Namespace(), "test-pvc", "1Gi")
```

Comment:
Where is the PVC deleted?

Reply:
Added removal. The cleanup slows down the test a bit, how bad is it to rely on the project teardown (in oc from util.CLI)?